classes/plugins/importexport/PKPImportExportDeployment.inc.php

```diff
@@ -121,6 +126,39 @@ function getSubmission() {
 	}
 
 	/**
+	 * Add the processed submission ID.
+	 * @param $submissionId integer
+	 */
+	function addProcessedSubmissionId($submissionId) {
+		$this->_processedSubmissionsIds[$submissionId] = array();
+	}
+
+	/**
+	 * Add the error message to the processed submission ID.
+	 * @param $submissionId integer
+	 * @param $errorMsg string
+	 */
+	function addSubmissionError($submissionId, $errorMsg) {
```

Comment:
Since we may also need to clean up other dependent types, maybe this would make a good general feature rather than just submission-specific. Maybe something like `addError($assocType, $assocId, $errorMsg)` etc? Likewise `set`/`getProcessedSubmissionsIds`.